api/commands/reminders/remind.js

```diff
@@ -41,34 +39,52 @@ module.exports = class RemindCommand extends Command {
   }
 
   run(msg, { target, content, datetime }) {
-    const author = {
-      username: msg.message.author.username,
-      discriminator: msg.message.author.discriminator
-    };
+    const { author } = msg.message;
 
-    let parsedTime;
+    let timezone;
+    let reminders = [];
+    let remindersJson;
 
     if (!chrono.parseDate(datetime)) {
       return new Promise((resolve, reject) =>
         reject(msg.say(exceptions.invalid_datetime_format)));
     }
 
-    return selectTz([author.username, author.discriminator])
-    .then((timezone) => {
-      if (!timezone) {
-        return Promise.reject(msg.say(exceptions.timezone_not_set));
+    return tzStore.getAsync(author.id)
+    .then((tz) => {
+      if (!tz) {
+        msg.say(exceptions.timezone_not_set);
+      } else {
+        timezone = tz;
       }
-
-      parsedTime = parseDate(datetime, timezone);
+    })
+    .then(() => {
+      const parsedTime = parseDate(datetime, timezone);
 
       if (parsedTime.delayAmt < 500) {
         return Promise.reject(msg.say(exceptions.past_time));
       }
 
-      return addToQueue(target, content, parsedTime, author);
+      return reminderStore.getAsync(parsedTime.timeInMS)
+      .then((existingReminders) => {
+        if (existingReminders) {
+          reminders = JSON.parse(existingReminders);
+        }
+
+        reminders.push({
+          target: target.id,
+          parsedTime: parsedTime.parsed,
+          content
+        });
+
+        remindersJson = JSON.stringify(reminders);
```

Comment:
why do we need to set this variable here?
it looks like it's only used in the next `then` clause so i think you can jsut return it, use it in the next chain part, and remove the `let` setting of it at the beginning of the chain